### PR TITLE
Using concurrency for tour reimport - PR 2

### PR DESCRIPTION
**Pull request by FJBDev**
- Updating the FIT SDK to 21.60

### CODE REVIEW

File: bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java [hunk 1]
```diff
@@ -2634,7 +2625,7 @@ private static boolean isTableAvailable(final Connection conn, final String tabl
     *           does not make sense to set the modified date.
     * @return persisted {@link TourData} or <code>null</code> when saving fails
     */
-   public static TourData saveTour(final TourData tourData, final boolean isUpdateModifiedDate) {
+   public static synchronized TourData saveTour(final TourData tourData, final boolean isUpdateModifiedDate) {
```
Comment (FJBDev) [inline bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java:2628]: Changing saveTour to `synchronized`

File: bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java [hunk 2]
```diff
@@ -2757,7 +2748,191 @@ public static TourData saveTour(final TourData tourData, final boolean isUpdateM
...
+
+         updateCachedFields(persistedEntity);
+
+         saveTour_GeoParts(persistedEntity);
+      }
+
+      return persistedEntity;
+   }
+
+   private static synchronized void saveTour_GeoParts(final TourData tourData) {
```
Comment (FJBDev) [inline bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java:2935]: Changing saveTour_GeoParts to `synchronized`

File: bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java [hunk 3]
```diff
@@ -2889,7 +3064,7 @@ public static void updateActiveTourTypeList(final TourTypeFilter tourTypeFilter)
       _activeTourTypes = new ArrayList<>();
    }
 
-   private static void updateCachedFields(final TourData tourData) {
+   private static synchronized void updateCachedFields(final TourData tourData) {
```
Comment (FJBDev) [inline bundles/net.tourbook/src/net/tourbook/database/TourDatabase.java:3067]: Changing updateCachedFields to `synchronized`

**Comment by wolfgang-ch** [comment]
I get this NPE when running a re-import with this PR

```
java.lang.reflect.InvocationTargetException
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:397)
	at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:468)
	at net.tourbook.ui.views.rawData.DialogReimportTours.actionReimportSelectedTours(DialogReimportTours.java:307)
	at net.tourbook.ui.views.rawData.DialogReimportTours.doReimport(DialogReimportTours.java:932)
	at net.tourbook.ui.views.rawData.DialogReimportTours.lambda$9(DialogReimportTours.java:1093)
	at net.tourbook.ui.views.rawData.DialogReimportTours$$Lambda$723/0x0000000000000000.run(Unknown Source)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:74)
	at net.tourbook.ui.views.rawData.DialogReimportTours.okPressed(DialogReimportTours.java:1060)
	at org.eclipse.jface.dialogs.Dialog.buttonPressed(Dialog.java:468)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:619)
	at org.eclipse.jface.dialogs.Dialog$$Lambda$124/0x0000000000000000.accept(Unknown Source)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:84)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4209)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1043)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4026)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3626)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at net.tourbook.ui.views.rawData.ActionReimportTours.run(ActionReimportTours.java:44)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:474)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:580)
	at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:414)
	at org.eclipse.jface.action.ActionContributionItem$$Lambda$377/0x0000000000000000.handleEvent(Unknown Source)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4209)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1043)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4026)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3626)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.ui.internal.Workbench$$Lambda$122/0x0000000000000000.run(Unknown Source)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
	at net.tourbook.application.Application.start(Application.java:35)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:654)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:591)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1462)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1435)
Caused by: java.lang.NullPointerException
	at net.tourbook.ui.views.rawData.DialogReimportTours$1.run(DialogReimportTours.java:291)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)

```

**Comment by FJBDev** [comment]
> I get this NPE when running a re-import with this PR
> 

I created a [new PR](https://github.com/wolfgang-ch/mytourbook/pull/379) as I forgot to check-in a file that caused this issue.

**Pull request merged**